util/src/main/java/org/killbill/billing/util/entity/dao/EntityDaoBase.java

```diff
@@ -221,6 +221,26 @@ public Iterator<M> build(final EntitySqlDao<M, E> sqlDao, final Long offset, fin
                                               context);
     }
 
+    @Override
+    public Pagination<M> getByAccountRecordId(final Long offset, final Long limit, final InternalTenantContext context) {
+        return paginationHelper.getPaginationWithAccountRecordId(realSqlDao,
+                                                                 new PaginationIteratorBuilder<M, E, EntitySqlDao<M, E>>() {
+                                                                     @Override
+                                                                     public Long getCount(final EntitySqlDao<M, E> sqlDao, final InternalTenantContext context) {
+                                                                         // Only need to compute it once, because no search filter has been applied (see DefaultPaginationSqlDaoHelper)
+                                                                         return null;
+                                                                     }
+
+                                                                     @Override
+                                                                     public Iterator<M> build(final EntitySqlDao<M, E> sqlDao, final Long offset, final Long limit, final Ordering ordering, final InternalTenantContext context) {
+                                                                         return sqlDao.getByAccountRecordIdWithPaginationEnabled(offset, limit, context); //TODO_1792: Is this correct or is a new query (similar to EntitySqlDao.sql.stg#Line147 but with accountRecordId) required? This query does not take into account the ordering parameter
```

Comment:
I'm trying to understand why this is needed / why it's not symmetrical to https://github.com/killbill/killbill/pull/1789. Is it because we go through the internal API?

Reply:
Good catch! [DefaultInvoiceDao.getInvoicesForAccount](https://github.com/killbill/killbill/blob/7ea3acf0a95a520991985904b682d7ddc8a34e48/invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java#L188) is no longer required as the same logic is written in `EntityDaoBase.getByAccountRecordId` (for subscription bundles) and can be used for invoices as well. I've now deleted the code from `InvoiceDao`.